### PR TITLE
feat(feedback): add assignment component for single feedbacks

### PR DESCRIPTION
**Pull request by michellewzhang**
Create an assignment dropdown component based on the issues one. We have to make a new dropdown since the issues style doesn't fit with the feedback style; a lot of the code and logic is copied over from `components/group/assignedTo.tsx` though.

Unassign functionality works too. 

https://github.com/getsentry/sentry/assets/56095982/4ca28371-bb2a-4db5-aa32-1f819b16a42c

### CODE REVIEW

File: static/app/components/feedback/feedbackItem/feedbackViewers.tsx [hunk 1]
```diff
@@ -16,20 +14,19 @@ export default function FeedbackViewers({feedbackItem}: Props) {
...
+      maxVisibleAvatars={13}
+      tooltipOptions={{position: 'top'}}
+      renderTooltip={user => (
+        <Fragment>
+          {userDisplayName(user)}
+          <br />
+          <DateTime date={(user as AvatarUser).lastSeen} />
+        </Fragment>
+      )}
+      alignLeft
```
Comment (ryan953) [inline static/app/components/feedback/feedbackItem/feedbackViewers.tsx:29]: We could probably avoid this by doing:
```
const StyledAvatarList = styled(AvatarList)`
 ...
`;
```
A one-off prop isn't really great to add to a base component like AvatarList, and this is more common around the app.

File: static/app/components/group/assignedTo.tsx [hunk 2]
```diff
@@ -158,7 +162,7 @@ export function getAssignedToDisplayName(group: Group) {
     return user?.name ?? group.assignedTo.name;
   }
 
-  return group.assignedTo?.name ?? t('No one');
+  return group.assignedTo?.name ?? isFeedback ? t('Unassigned') : t('No one');
```
Comment (ryan953) [inline static/app/components/group/assignedTo.tsx:165]: Lets move the `t('No one')` and `t('Unassigned')` translations into the callsites, then we don't need to pass in `isFeedback` at all.

So the code would look like this in all the places:
```
import {getAssignedToDisplayName} from '...';

return <span>{getAssignedToDisplayName(issue) ?? t('No one')}</span>;
```

this changes the function return type from `string` to `string | undefined` but de-couples the function from that string which is what we want.

File: static/app/components/feedback/feedbackItem/feedbackAssignedTo.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  gap: ${space(0.5)};
+  max-width: 150px;
+  line-height: 1;
+`;
+
+const ActorName = styled('div')`
+  line-height: 1.2;
+  ${p => p.theme.overflowEllipsis}
+  font-size: ${p => p.theme.fontSizeSmall};
+`;
```
Comment (ryan953) [inline static/app/components/feedback/feedbackItem/feedbackAssignedTo.tsx:126]: light styling to make it work on our page 👍

**Review by ryan953 — APPROVED** [review]

File: static/app/components/feedback/feedbackItem/feedbackAssignedTo.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+                actor={feedbackIssue.assignedTo}
+                hasTooltip={false}
+                size={16}
+              />
+            )}
+            <ActorName>{getAssignedToDisplayName(feedbackIssue, true)}</ActorName>
+            <IconChevron
+              data-test-id="assigned-to-chevron-icon"
+              direction={isOpen ? 'up' : 'down'}
+              size="sm"
```
Comment (billyvg) [inline static/app/components/feedback/feedbackItem/feedbackAssignedTo.tsx:103]: Let's get rid of the test-ids if we are not using them

Button could  benefit from an `aria-label` too

File: static/app/components/feedback/feedbackItem/feedbackAssignedTo.tsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    },
+    {
+      notifyOnChangeProps: ['data'],
+      enabled: defined(feedbackEvent?.id),
+    }
+  );
+
+  useEffect(() => {
+    // TODO: We should check if this is already loaded
+    fetchOrgMembers(api, organization.slug, [project.id]);
```
Comment (billyvg) [inline static/app/components/feedback/feedbackItem/feedbackAssignedTo.tsx:44]: Will this be called every time we load a new feedback?

File: static/app/components/feedback/feedbackItem/feedbackAssignedTo.tsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+      .then(response => {
+        if (unmounted) {
+          return;
+        }
+
+        setEventOwners(response);
+      });
+
+    return () => {
+      unmounted = true;
```
Comment (billyvg) [inline static/app/components/feedback/feedbackItem/feedbackAssignedTo.tsx:67]: Should we cancel the api request on unmount? `api.cancel()` iirc

Comment (michellewzhang) [reply static/app/components/feedback/feedbackItem/feedbackAssignedTo.tsx:68]: `api.clear()` maybe?

File: static/app/components/feedback/feedbackItem/feedbackAssignedTo.tsx (hunk 5 above)
Comment (michellewzhang) [reply static/app/components/feedback/feedbackItem/feedbackAssignedTo.tsx:45]: yes i think so 😔... but this is how issues does it

**Review by billyvg — APPROVED** [review]
imo we should wait to merge this until optimistic update is working, otherwise the UI looks broken (esp since we are launching for alpha soon).

**Comment by michellewzhang** [comment]
> imo we should wait to merge this until optimistic update is working, otherwise the UI looks broken (esp since we are launching for alpha soon).

yes that sounds good!

**Review by billyvg — APPROVED** [review]

**Pull request merged**